pkg/controller/autoscaling/elasticsearch/status/events.go

```diff
@@ -0,0 +1,29 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package status
+
+import (
+	"strings"
+
+	esv1 "github.com/elastic/cloud-on-k8s/pkg/apis/elasticsearch/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/record"
+)
+
+// EmitEvents emits a selected type of event on the Kubernetes cluster event channel.
+func EmitEvents(elasticsearch esv1.Elasticsearch, recorder record.EventRecorder, status Status) {
+	for _, status := range status.AutoscalingPolicyStatuses {
+		emitEventForAutoscalingPolicy(elasticsearch, recorder, status)
+	}
+}
+
+func emitEventForAutoscalingPolicy(elasticsearch esv1.Elasticsearch, recorder record.EventRecorder, status AutoscalingPolicyStatus) {
+	for _, event := range status.PolicyStates {
+		switch event.Type {
+		case VerticalScalingLimitReached, HorizontalScalingLimitReached, MemoryRequired, StorageRequired:
```

Comment:
Just curious why are we dropped certain event types here?

Reply:
I believe my initial thinking was that there may not be a 1-1 relationship between the events we want in the resource status _(primarily targeting the user)_ and the events we want to raise at the cluster level _(targeting the user and the K8S cluster admins)_.
For example I'm not sure that an empty response from Elasticsearch (`EmptyResponse`  event), which is likely to be a temporary situation, really deserves a K8S cluster event. While reaching a limit, on the contrary, is a critical situation which might require to be surfaced more broadly.
I agree it is not a strong argument, just a feeling, so I think I'm fine removing this filter for the moment.

Reply:
Makes sense, we can adjust this later as well.